components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -685,6 +686,11 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
             publishAuthenticationSuccess(request, context, sequenceConfig.getAuthenticatedUser());
         }
 
+        // Passing the federated tokens to the authentication result.
+        if (context.getProperty(FEDERATED_TOKENS) != null) {
+            authenticationResult.addProperty(FEDERATED_TOKENS, context.getProperty(FEDERATED_TOKENS));
```

Comment:
Lets add a debug log here mentioning federated tokens are available and being added to the auth result. Would be better if we can mention from which IDPs as well.

Reply:
Fixed cfb085b712df445a0a612df800c012b16fa75faf